llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -965,7 +965,7 @@ void VPInstruction::dump() const {
 
 void VPInstruction::print(raw_ostream &O, const Twine &Indent,
                           VPSlotTracker &SlotTracker) const {
-  O << Indent << "EMIT ";
+  O << Indent << (isSingleScalar() ? "SINGLE-SCALAR " : "EMIT ");
```

Comment:
Would
```suggestion
  O << Indent << (isSingleScalar() ? "EMIT-SCALAR " : "EMIT ");
```
be better, keeping "EMIT" as common prefix for VPInstructions?

Cf. "CLONE" vs. "REPLICATE" of VPReplicateRecipe, where an underlying Inst is replicated-once=cloned, or more.

Reply:
Or print it like a flag, something like `EMIT single-scalar vp<%foo> = ...`

Reply:
Thanks, I think both are better than SINGLE-SCALAR! I went with `EMIT-SCALAR` for now, as it is slightly shorter.